### DISCUSSION TITLE
python的序列问题跟随变化

### DISCUSSION THREAD

**Discussion by MeAPikachu** [physics-data/faq#48] [Q&A] [answered]
使用 a.append(b) 之后，如果b发生改变，那么a中的值也发生了改变；
我利用的是 元组的不变性质（自己想的..），
c=tuple(b), a.append(list(c)) 
有更好的解决方式嘛

**Comment by MeAPikachu** [comment]
(没注意到 有其他同学提出类似 问题..)

**Comment by Harry-Chen** [comment, marked as answer]
看 `b` 的类型，如果是可变类型（如 `list`, `dict`），在插入到列表时，如果需要不跟随改变，通常需要复制一份，有以下的方法：

* `a.append(list(b))`
* `a.append(b[:])`
* `a.append(copy.copy(b))` （参见 <https://docs.python.org/zh-cn/3/library/copy.html>）

你的做法结果是正确的，但是理解是不对的，其实是因为 tuple 和 list 的构造过程中都会复制一遍原来的元素。
